### ISSUE TITLE
Please cut down on the frequency of updates

### ISSUE THREAD

**Issue by ronbarr** [dmarmor/epichrome#257]
I have 16 epichrome apps that are each frequently asking me to update. Can we bundle a number of changes and only do major releases through the update mechanism?

**Comment by henderea** [comment]
My personal suggestion would be to have different “channels”, with one for major features and one for every point release. I definitely don’t want to have the updates come less often for myself, so I think it would be best to allow users to select the update frequency. 

That said, I think the recent bunch of release is related to the recent significant rework of the software. I’m not too sure it will _usually_ be this frequent.

**Comment by dmarmor** [comment]
Thanks for the suggestion @ronbarr. And yes, @henderea, I agree, the solution is not fewer updates. Part of what's going on here is that with the introduction of the built-in engine I need to try to keep up with new Brave releases, which seem to happen about weekly, if not more frequently. Here's my thought (which I won't be able to introduce until 2.4):

1. Change the way checking for updates on GitHub works so that once one app pops up the dialog letting the user know there's a new version out there, no other Epichrome apps will check for that version again. You'll only ever see that dialog once per version.

2. Add a per-app setting that sets whether that app should check for all updates on GitHub, or only important and major ones. If you set all your apps to that setting, none of them will pop up a dialog when a minor (e.g. Brave version-only) update goes live. Note that if you run Epichrome.app itself, that will always still check for even minor updates.

Let me know what you think of that as a solution. Thanks!

**Comment by ronbarr** [comment]
@dmarmor That sounds great. Thanks for all of your hard work.

**Comment by talleux** [comment]
@dmarmor could there be a possibility to udpate all apps at once? I appreciate the frequent updates as they all are improvements, but having 30 apps (20 of wich i use on a daily basis) i'd love to press a "update all" button...

**Comment by dmarmor** [comment]
@talleux Yes! That's going to be possible with 2.4.0 as part of app-editing. You'll be able to drop all your apps onto Epichrome.app and tell it to update all of them at once.

I'm also thinking about adding an option at app-creation (or during editing) to make an app auto-update. Any app with that setting would update itself when it detects a new version of Epichrome installed, without asking first. Would that be useful to you?

**Comment by demmer** [comment]
Strong +1 to the idea of auto-updating for minor releases.

As I mentioned in #279 I have to consistently update to access internal sites as Duo 2FA blocks access from out of date browsers, so making this automatic would be even better.

**Comment by dmarmor** [comment]
Yes! Auto-updating will be an option during app creation and editing for 2.4 apps. The feature is already in the latest beta.

**Comment by cdeutsch** [comment]
In the meantime is there a simple way to turn off update checks?

Literally every time I open an Epichrome app it's nagging me to update.

**Comment by dmarmor** [comment]
Until 2.4.0 there's not a simple way to turn off updates. The best I can offer (and it will be a bit of work if you have a lot of apps) would be this:

To silence GitHub checks in an app:
1. Make sure your app is _not_ running.
2. Open the app's `config.sh` file, found in `~/Library/Application Support/Epichrome/Apps/<AppID>/config.sh`.
3. You will see a line like this: `SSBUpdateCheckDate='1613326059'`. Edit that line to add an extra 0 at the end of the number, so it now looks like this: `SSBUpdateCheckDate='16133260590'`
4. If you also want to stop an app from offering to update itself when you install a new 2.3.x version on your system, also edit this line: `SSBUpdateVersion='2.3.25'` to this `SSBUpdateVersion='2.3.99'`
5. Save your changes.

This will suppress GitHub warnings pretty much indefinitely (for about 460 years anyway). If you also did step 4, it will also stop your app from trying to update to any future 2.3.x version you install. If you ever want to undo any of this, you can just re-edit the config.sh.

I hope this helps! Version 2.4.0 is coming quite soon, and has much more control over app update checks...
